app/react/hierarchy/keys/index.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import React from 'react';
+import preload from 'redux-preload';
+import { connect } from 'react-redux';
+
+import createWaitFor from '../../utils/waitFor';
+import { fetchHierarchy } from '../actions';
+import Key from './Key';
+
+const preloader = <div>Preloading</div>;
+const waitFor = createWaitFor(preloader);
+
+const transformHierarchy = (structure: Object, level: number = 0): Array<Object> =>
```

Comment:
Just to be safe, there should be max level check to prevent long cycling in case you obtain malformed, huge, and nested object.